sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/models/ListBatchNodesOptions.java

```diff
@@ -0,0 +1,68 @@
+package com.azure.compute.batch.models;
+
+import java.util.List;
+
+public class ListBatchNodesOptions extends BatchBaseOptions{
+    private String filter;
+    private Integer maxresults;
+    private List<String> select;
+
+    /**
+     * Gets the OData $filter clause used for filtering results.
+     *
+     * @return The OData $filter clause.
+     */
+    public String getFilter() {
+        return filter;
+    }
+
+    /**
+     * Sets the OData $filter clause used for filtering results.
+     *
+     * @param filter The OData $filter clause.
+     */
+    public void setFilter(String filter) {
```

Comment:
To be consistent with how setters on models work, these should return `this`. That enables code that looks like this:

```
new ListBatchNodesOptions()
    .setSelect(Arrays.asList("id", "state"))
    .setFilter("state eq 'idle'");
```